client/my-sites/checkout/composite-checkout/components/wp-checkout-order-review.tsx

```diff
@@ -115,7 +112,12 @@ export default function WPCheckoutOrderReview( {
 		return () => {
 			shouldCheck = false;
 		};
-	}, [ isDesktop ] );
+	}, [] );
+
+	function retrieveExperimentName() {
+		const cookies = cookie.parse( document.cookie );
```

Comment:
The only thing that worries me about this is: what happens if cookie access is unavailable? Will this throw an Error in that case? What happens if `cookie.parse()` fails? Its documentation does not seem to make that clear. 

The reason this is concerning to me is not for this particular flow, but because the above `useEffect` will run `retrieveExperimentName()` every time that checkout loads.

If you're not certain about failure cases, my instinct would be to wrap the interior of this function in `try/catch` and log the errors, which should allow the rest of checkout to work even if there are unexpected failures here. That way we make checkout as forgiving of browser errors as possible.

Reply:
Thanks Payton, I added a try/catch block to the cookie read per your suggestion.